### PR TITLE
User/dhoehna/fixing theming on content dialogs

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
This relates to content dialogs with content made from an adaptive card.
1. The adaptive card will not listen to the DevHome theme.  If Windows is in light mode and DevHome is in dark mode then the dialog box is in light mode while the adaptive card content is in dark mode.
2. The content dialog box will change themes when open.

This PR is a work around.  The work around is
1. Re-get the renderer.
3. Re-get the adaptive card
4. Re-assign the content of a dialog box to the adaptive car content retrieved earlier.

This issue only appears with content dialog boxes where the content is an adaptive card.

## References and relevant issues

https://task.ms/51611819

## Detailed description of the pull request / Additional comments

The only two instances I know of in DevHome are
1. Environments
2. QuickStart Playground.

Other instances will need to change over manually.

## Validation steps performed
Manual.

In addition to the two videos I tested changing the windows theme while DevHome's theme is set to Dark.  As expected the content dialog did not change theme.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated
- [ ] Telemetry [compliance tasks](https://aka.ms/devhome-telemetry) completed for added/updated events

Movies!

Windows in light theme and DevHome in dark.
https://github.com/user-attachments/assets/60e48be1-dc5e-4d4a-a6b1-b9b3dde71573

Changing windows themes with DevHome theme set to "Windows Default"
https://github.com/user-attachments/assets/83327aa4-f5ad-489d-8650-4d822de749a7

### CODE REVIEW

File: common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs [hunk 1]
```diff
@@ -15,7 +17,11 @@ namespace DevHome.Common.Views.AdaptiveCardViews;
 /// </summary>
 public sealed partial class ContentDialogWithNonInteractiveContent : ContentDialog
 {
-    public ContentDialogWithNonInteractiveContent(DevHomeContentDialogContent content)
+    private readonly IThemeSelectorService _themeSelector;
+
+    private readonly DevHomeContentDialogContent _content;
+
+    public ContentDialogWithNonInteractiveContent(DevHomeContentDialogContent content, IThemeSelectorService themeSelector)
```
[the remark this answers is not in the document]
Comment (dhoehna) [reply common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs]: Done.

File: common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs [hunk 2]
```diff
@@ -15,27 +17,57 @@ namespace DevHome.Common.Views.AdaptiveCardViews;
...
+        _themeSelector.ThemeChanged += OnThemeChanged;
+        _content = content;
+    }
+
+    private async void OnThemeChanged(object? sender, ElementTheme newRequestedTheme)
+    {
+        // set the theme of the content dialog box.
+        RequestedTheme = newRequestedTheme;
+        var cardContent = await MakeCardContentAsync();
+        Content = new ScrollViewer
```
Comment (krschau) [inline common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs:58]: Would it be easier for MakeCardContent() to return this ScrollViewer, so you don't have to remake it in a couple places?

File: common/DevHomeAdaptiveCards/CardModels/DevHomeLaunchContentDialogButton.cs [hunk 3]
```diff
@@ -7,7 +7,10 @@
 using AdaptiveCards.ObjectModel.WinUI3;
 using CommunityToolkit.Mvvm.Input;
 using DevHome.Common.DevHomeAdaptiveCards.CardInterfaces;
+using DevHome.Common.Extensions;
 using DevHome.Common.Views.AdaptiveCardViews;
+using DevHome.Contracts.Services;
+using Microsoft.UI.Xaml;
```
[the remark this answers is not in the document]
Comment (dhoehna) [reply common/DevHomeAdaptiveCards/CardModels/DevHomeLaunchContentDialogButton.cs:13]: You are correct.  Removed.

File: common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs (hunk 2 above)
Comment (dhoehna) [reply common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs]: Technically, yes.  De-duping is always good.  Done.

File: common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs [hunk 4]
```diff
@@ -15,27 +17,53 @@ namespace DevHome.Common.Views.AdaptiveCardViews;
...
-            {
-                VerticalScrollBarVisibility = ScrollBarVisibility.Auto,
-                Content = card.FrameworkElement,
-            };
+            var cardContent = await MakeCardContentAsync();
+            Content = await MakeCardContentAsync();
+
+            // Set the theme of the content dialog box
+            RequestedTheme = _themeSelector.IsDarkTheme() ? ElementTheme.Dark : ElementTheme.Light;
             SecondaryButtonText = content.SecondaryButtonText;
```
Comment (krschau) [inline common/Views/AdaptiveCardViews/ContentDialogWithNonInteractiveContent.xaml.cs:41]:
```suggestion
            Content = await MakeCardContentAsync();

            // Set the theme of the content dialog box
            RequestedTheme = _themeSelector.IsDarkTheme() ? ElementTheme.Dark : ElementTheme.Light;
            SecondaryButtonText = Content.SecondaryButtonText;
```

**Review by krschau — APPROVED** [review]

**Review by guimafelipe — APPROVED** [review]

**Pull request merged**